extras/ams_wrapper/src/api/types.py

```diff
@@ -0,0 +1,129 @@
+#
+# Copyright (c) 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+from typing import List
+from abc import ABC
+
+class Tag:
+    def __init__(self, value: str, confidence: float):
+        self.value = value
+        self.confidence = confidence
+
+    def as_dict(self):
+        result_dict = {
+            "value": self.value,
+            "confidence": self.confidence
+        }
+        return result_dict
```

Comment:
How about `return vars(self)` ? We could potentially avoid defining attributes both in `__init__` and `as_dict` that way.

Reply:
I was thinking of a single point (one method name) for all types that would convert them to dict type. This explicit dictionary definition is consistent among all types classes. 
Your idea is cool for simple types, but for example for Entity I don't put entities list as is, but I do:  `"entities": [entity.as_dict() for entity in self.entities]` and I couldn't do just 
`return vars(self)` here right?

Also, if in the future types objects will require some internal fields that shall not be included in output dictionary or naming change will be required then we would still have to define output dict explicitly.